pydatastructs/trees/tests/test_binary_trees.py

```diff
@@ -319,3 +319,21 @@ def test_BinaryIndexedTree():
     assert t.get_sum(0, 2) == 105
     assert t.get_sum(0, 4) == 114
     assert t.get_sum(1, 9) == 54
+
+def test_issue_234():
+    """
+    https://github.com/codezonediitj/pydatastructs/issues/234
+    """
+    tree = SelfBalancingBinaryTree() 
+    tree.insert(5,5)
+    tree.insert(5.5,5.5)
+    tree.insert(4.5,4.5)
+    tree.insert(4.6,4.6)
+    tree.insert(4.4, 4.4)
+    tree.insert(4.55, 4.55)
+    tree.insert(4.65, 4.65)
+    
+    tree._right_rotate(3, 5)
+    
+    assert tree.tree[3].parent == 5
+    assert tree.tree[2].right != 3
```

Comment:
Check the full tree, `str(tree)`.
May be, testing on the mirror image of the tree will test the lines marked in red [here](https://codecov.io/gh/codezonediitj/pydatastructs/pull/238/diff).
Can you please try that out?

Reply:
I have added the full tree check as well and also for the right child

Reply:
Added checks for `_left_rotate` as well for coverage